packages/core/core/src/Dependency.js

```diff
@@ -85,12 +85,29 @@ export function createDependency(
 }
 
 export function mergeDependencies(a: Dependency, b: Dependency): void {
-  let {meta, symbols, ...other} = b;
+  let {
+    meta,
+    symbols,
+    needsStableName,
+    isEntry,
+    isOptional,
+    bundleBehavior,
+    ...other
+  } = b;
   Object.assign(a, other);
   Object.assign(a.meta, meta);
   if (a.symbols && symbols) {
     for (let [k, v] of symbols) {
       a.symbols.set(k, v);
     }
   }
+  if (needsStableName) a.needsStableName = true;
+  if (isEntry) a.isEntry = true;
+  if (!isOptional) a.isOptional = false;
+  if (
+    bundleBehavior == BundleBehavior['isolated'] ||
+    a.bundleBehavior == null
+  ) {
+    a.bundleBehavior = bundleBehavior;
```

Comment:
@wbinnssmith this has implications for the bundling issue we were discussing. If we have two dependencies on the same asset, one isolated and one not, would we want two different bundles or should they be merged?

Reply:
I think there are arguments for each: one would produce fewer overall bundles and there would be one canonical bundle for a given entry asset, and the other would produce multiple bundles, but each with its own optimal set of assets (non-isolated bundles can rely on existing loaded assets).

Reply:
I think we could understand what is "best" by real-world tests.